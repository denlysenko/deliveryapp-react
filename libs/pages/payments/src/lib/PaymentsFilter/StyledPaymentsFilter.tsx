import styled from 'styled-components';

export const StyledPaymentsFilter = styled.div`
  color: pink;
`;
