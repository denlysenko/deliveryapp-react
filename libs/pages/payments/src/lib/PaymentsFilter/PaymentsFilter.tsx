import React from 'react';

import { StyledPaymentsFilter } from './StyledPaymentsFilter';

/* eslint-disable-next-line */
export interface PaymentsFilterProps {}

export const PaymentsFilter = (props: PaymentsFilterProps) => {
  return (
    <StyledPaymentsFilter>
      <h1>Welcome to PaymentsFilter component!</h1>
    </StyledPaymentsFilter>
  );
};
