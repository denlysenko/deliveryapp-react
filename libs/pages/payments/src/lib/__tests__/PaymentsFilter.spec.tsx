import React from 'react';

import { fireEvent, render, screen } from '@testing-library/react';

import { PaymentsFilter } from '../PaymentsFilter/PaymentsFilter';

jest.mock('lodash', () => ({
  ...jest.requireActual<any>('lodash'),
  debounce: jest.fn((fn) => fn)
}));

const handleFilterChange = jest.fn();

describe('PaymentsFilter', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render successfully', () => {
    const { baseElement } = render(
      <PaymentsFilter handleFilterChange={handleFilterChange} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render empty string for searchTerm if filter prop is not present', () => {
    render(<PaymentsFilter handleFilterChange={handleFilterChange} />);
    expect(screen.getByPlaceholderText(/search by number/i)).toHaveValue('');
  });

  it('should render empty string for searchTerm if filter prop is empty', () => {
    render(
      <PaymentsFilter filter={{}} handleFilterChange={handleFilterChange} />
    );
    expect(screen.getByPlaceholderText(/search by number/i)).toHaveValue('');
  });

  it('should render searchTerm from filter', () => {
    render(
      <PaymentsFilter
        filter={{ id: 1 }}
        handleFilterChange={handleFilterChange}
      />
    );
    expect(screen.getByPlaceholderText(/search by number/i)).toHaveValue('1');
  });

  describe('handleFilterChange', () => {
    it('should call if searchTerm has changed', () => {
      render(
        <PaymentsFilter filter={{}} handleFilterChange={handleFilterChange} />
      );

      fireEvent.change(screen.getByPlaceholderText(/search by number/i), {
        target: {
          value: '2'
        }
      });

      expect(handleFilterChange).toBeCalledTimes(1);
      expect(handleFilterChange).toBeCalledWith({ id: 2 });
    });
  });
});
