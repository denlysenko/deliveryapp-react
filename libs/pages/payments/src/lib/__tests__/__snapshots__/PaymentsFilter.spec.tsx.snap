// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentsFilter should render successfully 1`] = `
<body>
  <div>
    <div
      class="sc-AxjAm cDumgh"
    >
      <div
        class="p-inputgroup p-col-12 p-md-6"
      >
        <span
          class="p-inputgroup-addon"
        >
          <i
            class="fa fa-search"
          />
        </span>
        <input
          class="p-inputtext p-component"
          data-testid="searchTerm"
          placeholder="Search by number..."
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</body>
`;
